presto-spi/pom.xml

```diff
@@ -88,7 +88,7 @@
         <dependency>
             <groupId>com.google.guava</groupId>
             <artifactId>guava</artifactId>
-            <scope>test</scope>
+            <scope>provided</scope>
```

Comment:
noob question, why this change?

Reply:
Used com.google.common.collect.ImmutableList etc in CteConsumer which is defined in presto-spi. Before this guava was not used in presto-spi

Reply:
Let's not add in Guava or presume it's needed in the SPI--let's figure out another way (unmodifiableCollection perhaps)

Reply:
Any benefit in doing so?

Reply:
I think we need to add as little dependency as possible for SPI, let's remove the dependency here.

Reply:
I think I am not adding a new dependency here, I am just marking it provided, which means that its already available at runtime and just marking it available during compile time. What do you think? If not I can remove

Reply:
The problem with marking it provided is people will need to supply this dependency independently.  Suppose someone is creating a custom plugin--if their provided dependency conflicts with the one expected by the compiler, then it will generate runtime errors.  It's best to minimize SPI dependencies as much as possible.

Reply:
Okay, i will remove it